src/components/Colors/ColorContext.js

```diff
@@ -1,7 +1,53 @@
-import React from 'react'
-import colors from '../../theme/colors'
+import React, { useEffect, useState } from 'react'
+import colors, { variableColorKeys } from '../../theme/colors'
 
 const defaultColors = colors
 const ColorContext = React.createContext(defaultColors)
 
+export const generateCSSColorDefinitions = colors => {
+  return variableColorKeys
+    .map(key => `--color-${key}: ${colors[key]};`)
+    .join(' ')
+}
+
+// ensure only main colors are available via context
+const reduceMainColors = (mapper = key => key) =>
+  variableColorKeys.reduce((c, key) => {
+    c[key] = mapper(key)
+    return c
+  }, {})
+
+export const ColorContextProvider = ({ value: forcedValue, children }) => {
+  // we initially assume browser support it
+  // - e.g. during server side rendering
+  const [CSSSupport, setCSSSupport] = useState(true)
+  useEffect(() => {
+    let support
+    try {
+      support =
+        window.CSS &&
+        window.CSS.supports &&
+        window.CSS.supports('color', 'var(--primary)')
+    } catch (e) {}
+    if (!support) {
+      // but if can't confirm the support in the browser we turn it off
+      setCSSSupport(false)
+    }
+  }, [])
+
+  let colorValue
+  if (forcedValue) {
+    colorValue = reduceMainColors(key => forcedValue[key])
+  } else if (CSSSupport) {
+    colorValue = reduceMainColors(key => `var(--color-${key})`)
+    colorValue.rawColors = colors
```

Comment:
Shouldn't this be `colorValue.rawColors = reduceMainColors(key => colors[key])`? Why do we need a mapping to the global color object? In general I'd tend to invert those 2, i.e.:
```
colorValue.cssVars = reduceMainColors(key => `var(--color-${key})
colorValue = colors // reduceMainColors(key => colors[key])
```
That way the structure of the data is consistent through the 3 if clauses.